src/praisonai-train/praisonai_train/train/llm/trainer.py

```
@@ -550,12 +812,25 @@ def process_dataset(self, dataset_info):
+                "('conversations'), Alpaca ('instruction'/'input'/'output') and "
+                "raw text ('text'). Rename your columns to one of those shapes, "
+                "or set chat_template to match your data.")
         return dataset
 
     def load_datasets(self):
         datasets = []
         for dataset_info in self.config["dataset"]:
+            # Advertised in the shipped templates but never read. Saying so beats
+            # silently discarding a formatter the user believed was running.
+            if dataset_info.get("processing_func"):
+                logger.warning(
```

Ruff (0.16.2)

[error] Undefined name `logger`

(F821)